### PR TITLE
Add dynamic yaml example to MissingAnnotation component

### LINKED ISSUE #18264: 🚀 Feature: MissingAnnotationEmptyState should allow customized entity template
### 🔖 Feature description

Update the COMPONENT_YAML_TEMPLATE for `<MissingAnnotationEmptyState>` so that it can set the Entity kind and type based on the entity that is currently being viewed.

### 🎤 Context

When utilizing this react component to alert the user that they should add an annotation, it uses a baked in template to represent the entity that should be modified.

```
const COMPONENT_YAML_TEMPLATE = `apiVersion: backstage.io/v1alpha1
kind: Component
metadata:
  name: example
  description: example.com
  annotations:
    ANNOTATION: value
spec:
  type: website
  lifecycle: production
  owner: user:guest`;
```

This ends up looking odd if you are viewing a page that is not a component of type website, and can confuse or mislead users.

image

### ✌️ Possible Implementation

A few options:

1. Allow for just entity kind and type to be optionally passed in and override the template
2. Base the template dynamically on the entity being viewed using the `useEntity()` hook

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by bforbis**
## Hey, I just made a Pull Request!

The `MissingAnnotationEmptyState` component will now be aware of the entity being viewed and use the details of the current entity to generate the YAML snippet that includes the annotations helper on it.

image

Closes #18264

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by bforbis** [comment]
Not stale

**Comment by bforbis** [comment]
Still waiting on review

**Comment by bforbis** [comment]
Still waiting on review

**Comment by bforbis** [comment]
Still waiting on review

**Comment by bforbis** [comment]
@Rugvip can you review this?

File: packages/core-components/src/components/EmptyState/MissingAnnotationEmptyState.tsx
Comment (Rugvip) [inline packages/core-components/src/components/EmptyState/MissingAnnotationEmptyState.tsx:62]: To be honest I don't think we should list the existing annotations. For many catalogs it'll make the example extremely verbose 😅 It'll also make this logic quite a lot simpler

**Review by Rugvip — CHANGES_REQUESTED** [review]
Sorry, didn't realize this was failing for unrelated reasons 🤦 

Thank you, I like the idea! 👍 

👀  @backstage/catalog-maintainers as well

File: packages/core-components/src/components/EmptyState/MissingAnnotationEmptyState.tsx
Comment (bforbis) [reply packages/core-components/src/components/EmptyState/MissingAnnotationEmptyState.tsx:62]: This doesn't list all existing annotations, only the ones passed down in the `props` for `MissingAnnotationEmptyState` component. In most cases, that will just be one annotation.

As an example:

This would render
```ts
<MissingAnnotationEmptyState
    annotation=["db-type", "db-version"]
    readMoreUrl="/docs/default/component/backstage"
  />
```
image

Comment (bforbis) [reply packages/core-components/src/components/EmptyState/MissingAnnotationEmptyState.tsx:62]: @Rugvip is this explanation ok?

Comment (Rugvip) [reply packages/core-components/src/components/EmptyState/MissingAnnotationEmptyState.tsx:63]: Yep sounds good! 👍

**Review by Rugvip — APPROVED** [review]
Alright, makes sense! 👍 

Fixed the conflicts for you, let's :shipit:

**Pull request merged**
